src/ngx_http_datadog_module.cpp

```diff
@@ -736,6 +762,27 @@ char *merge_script(ngx_conf_t *conf, NgxScript &previous, NgxScript &current,
 
 }  // namespace
 
+//------------------------------------------------------------------------------
+// merge_datadog_srv_conf
+//------------------------------------------------------------------------------
+static char *merge_datadog_srv_conf(ngx_conf_t *cf, void *parent,
+                                    void *child) noexcept {
+  auto parent_srv_conf = static_cast<datadog_srv_conf_t *>(parent);
+  auto child_srv_conf = static_cast<datadog_srv_conf_t *>(child);
+
+  if (!child_srv_conf->service_name) {
+    child_srv_conf->service_name = parent_srv_conf->service_name;
+  }
+  if (!child_srv_conf->service_env) {
+    child_srv_conf->service_env = parent_srv_conf->service_env;
+  }
+  if (!child_srv_conf->service_version) {
+    child_srv_conf->service_version = parent_srv_conf->service_version;
```

Comment:
Can't we use ngx_conf_merge_value here ?

Reply:
`ngx_conf_merge_value` only works for ngx types.